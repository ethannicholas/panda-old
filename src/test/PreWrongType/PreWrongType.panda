class Test {
    @pre(i > 10)
    method test(i:Int) {
        Console.writeLine(i)
    }
}

class Test2 : Test {
    @override
    @pre(i > 0)
    method test(i:Int) {
        Console.writeLine(i)
    }
        
    @class
    @preOr(i < 0)
    method test2(i:Int) {
        Console.writeLine(i)
    }
}

var t := Test2()
t.test(5)
Test2.test2(-5)
