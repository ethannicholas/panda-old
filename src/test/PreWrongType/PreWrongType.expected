error: PreWrongType.panda:9:5: @pre may not be used together with @override; did you mean to use @preOr?
error: PreWrongType.panda:15:5: @preOr must be used together with @override
2 errors
